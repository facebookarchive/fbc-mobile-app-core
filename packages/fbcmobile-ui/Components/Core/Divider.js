/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import type {ViewStyleProp} from 'react-native/Libraries/StyleSheet/StyleSheet';

import * as React from 'react';
import {Colors} from '@fbcmobile/ui/Theme';
import {StyleSheet, View} from 'react-native';

type Props = {
  +style?: ViewStyleProp,
};

const Divider = ({style}: Props) => {
  return <View style={[styles.root, style]} />;
};

const styles = StyleSheet.create({
  root: {
    height: 1,
    backgroundColor: Colors.Divider,
  },
});

export default Divider;
