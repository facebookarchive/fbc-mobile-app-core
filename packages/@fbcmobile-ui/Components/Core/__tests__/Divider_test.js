/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import Divider from '../Divider';
import React from 'react';
import renderer from 'react-test-renderer';

test('Divider renders correctly', () => {
  const tree = renderer.create(<Divider />).toJSON();
  expect(tree).toMatchSnapshot();
});
