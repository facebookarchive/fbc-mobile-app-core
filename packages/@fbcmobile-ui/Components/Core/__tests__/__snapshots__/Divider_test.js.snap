// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#CFD1D5",
        "height": 1,
      },
      undefined,
    ]
  }
/>
`;
